### PR TITLE
Feat/msdk 665 paying key balance

### PR DESCRIPTION
**Pull request by monitz87**

### CODE REVIEW

File: src/base/TransactionQueue.ts [hunk 1]
```diff
@@ -363,4 +425,61 @@ export class TransactionQueue<
...
+        noThirdPartyAllowance.push(data);
+      }
+    });
+
+    if (noThirdPartyAllowance.length) {
+      throw new PolymeshError({
+        code: ErrorCode.ValidationError,
+        message: "Not enough POLYX third party allowance to pay for this procedure's fees",
+        data: {
+          accounts: noThirdPartyAllowance,
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/base/TransactionQueue.ts:458]: They are arrays of `ThirdPartyFees`, so they come with all that data already. Agreed that the name isn't the best

File: src/base/TransactionQueue.ts [hunk 2]
```diff
@@ -363,4 +425,61 @@ export class TransactionQueue<
...
+        },
+      });
+    }
+    if (noThirdPartyBalance.length) {
+      throw new PolymeshError({
+        code: ErrorCode.ValidationError,
+        message: "Not enough POLYX third party balance to pay for this procedure's fees",
+        data: {
+          accounts: noThirdPartyBalance,
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/base/TransactionQueue.ts:467]: They are arrays of `ThirdPartyFees`, so they come with all that data already. Agreed that the name isn't the best

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**